test/lit/passes/optimize-instructions-gc.wast

```diff
@@ -3120,4 +3120,67 @@
       (ref.null nofunc)
     )
   )
+
+  ;; CHECK:      (func $ref-cast-heap-type (type $ref?|$B|_ref|$B|_=>_none) (param $null-b (ref null $B)) (param $b (ref $B))
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (local.get $b)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (ref.as_non_null
+  ;; CHECK-NEXT:    (local.get $null-b)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
```

Comment:
Do we refinalize when we do this optimization since the result becomes more refined?

Reply:
Yes, there is a refinalize in this PR.